server/aesdsocket.c

```diff
@@ -290,12 +293,35 @@ void* thread_routine(void *arg)
 		if(NULL == routine_values->d_buf)
 		{
 			syslog(LOG_ERR,"Error: realloc()");
-            //free(routine_values->d_buf);
             error_handler();
             exit(ERROR_CODE);
 		}
     }
 
+    /* Logic for checking the IOCTL command : A9*/
+    char *tmp_buff = routine_values->d_buf;
+    char *ioctl_string =  "AESDCHAR_IOCSEEKTO:";
+    bool ioctl_received = false;
+    /* Compare the buffer we received from client with the ioctl command and see if they match */
+    if(0 == strncmp(tmp_buff, ioctl_string, strlen(ioctl_string)))
+    {
+        /* Line by line explanation
+        *  Create a seek that can store write cmd and offset
+        *  Point to the seek members i.e., write_cmd and offset, by adding command lenght to received string 
+        * AESDCHAR_IOCSEEKTO:X,Y += AESDCHAR_IOCSEEKTO will result in pointing to X,Y i.e,  write_cmd and offset.
+        * Using sscanf we can parse the tmp_buf using comma ',' as the seperator and get X and Y (write_cmd and offset) values seperately.
+        * Set the ioctl received flag to true. 
+        */
+        struct aesd_seekto seek;            
+        tmp_buff += strlen(ioctl_string);    
+        sscanf(tmp_buff, "%d,%d", &seek.write_cmd, &seek.write_cmd_offset);
```

Comment:
@Rishikesh38 what happens if sscanf fails?

Reply:
If sscanf fails, the ioctl call that follows may use incorrect or uninitialized values for seek.write_cmd and seek.write_cmd_offset. This could lead to unexpected behavior or incorrect results when calling ioctl. To handle the case where sscanf fails, I should have added some error checking